### PR TITLE
Feature/spk 90

### PR DESCRIPTION
**Pull request by AlexMisiulia**

### CODE REVIEW

File: app/src/main/java/com/simplifire/yotinfcclient/adapter/FormFieldAdapter.kt [hunk 1]
```diff
@@ -31,27 +30,15 @@ class FormFieldAdapter : ListAdapter<UiPersonalFormItem, FormFieldAdapter.FieldV
...
     }
 
     class FieldViewHolder(override val containerView: View)
         : RecyclerView.ViewHolder(containerView), LayoutContainer {
 
-        fun bind(formField: UiPersonalFormItem, isFirst: Boolean, isLast: Boolean) {
+        fun bind(formField: UiPersonalFormItem) {
             fieldNameTextView.text = formField.fieldName
             fieldValueTextView.text = formField.fieldValue
-
```
Comment (cquemin) [inline app/src/main/java/com/simplifire/yotinfcclient/adapter/FormFieldAdapter.kt]: Did you see how many time you had to do the same modification. This is usually a sign that there is too much duplication and the logic should be mutualised if possible

File: app/src/main/java/com/simplifire/yotinfcclient/fragment/PersonalFormFragment.kt [hunk 2]
```diff
@@ -45,6 +46,8 @@ class PersonalFormFragment : Fragment() {
 
     private fun initView() {
         fieldsRecyclerView.layoutManager = LinearLayoutManager(requireActivity(), RecyclerView.VERTICAL, false)
+        val borderDecorator = createBorderDecorator(requireActivity())
```
Comment (cquemin) [inline app/src/main/java/com/simplifire/yotinfcclient/fragment/PersonalFormFragment.kt]: Same comment about code duplication

File: app/src/main/java/com/simplifire/yotinfcclient/adapter/FormFieldAdapter.kt (hunk 1 above)
Comment (AlexMisiulia) [reply app/src/main/java/com/simplifire/yotinfcclient/adapter/FormFieldAdapter.kt]: I removed duplication to separate class. Or maybe I can't understand you

File: app/src/main/java/com/simplifire/yotinfcclient/fragment/PersonalFormFragment.kt (hunk 2 above)
Comment (AlexMisiulia) [reply app/src/main/java/com/simplifire/yotinfcclient/fragment/PersonalFormFragment.kt]: Not sure that I understand you. This is just borderDecorator creation. Do you mean this line is code duplication?

File: app/src/main/java/com/simplifire/yotinfcclient/adapter/FormFieldAdapter.kt (hunk 1 above)
Comment (cquemin) [reply app/src/main/java/com/simplifire/yotinfcclient/adapter/FormFieldAdapter.kt]: No you are right you what's left is not duplicated my bad, I mis read.

File: app/src/main/java/com/simplifire/yotinfcclient/fragment/PersonalFormFragment.kt (hunk 2 above)
Comment (cquemin) [reply app/src/main/java/com/simplifire/yotinfcclient/fragment/PersonalFormFragment.kt]: I was saying the `initView` method in `PersonalFormFragment` and in `FormDetailFragment` looks very similar.

Comment (AlexMisiulia) [reply app/src/main/java/com/simplifire/yotinfcclient/fragment/PersonalFormFragment.kt]: Yes, but In my opinion, this is the case where copy pasting 3 lines of code
```
        fieldsRecyclerView.layoutManager = LinearLayoutManager(requireActivity(), RecyclerView.VERTICAL, false)
        val borderDecorator = createBorderDecorator(requireActivity())
        fieldsRecyclerView.addItemDecoration(borderDecorator)
```

 is better than create abstraction like `initBorderRecyclerView`

Comment (cquemin) [reply app/src/main/java/com/simplifire/yotinfcclient/fragment/PersonalFormFragment.kt]: If that is the only logic in common then i agree. It was way for me to raise the question: does it make sense to put the logic in common.

**Pull request merged**